docs/deploy-rbd.md

```diff
@@ -280,6 +280,34 @@ Configure a role(s) for service accounts used for ceph-csi:
 * nodeplugin service account (`rbd-csi-nodeplugin`) requires **create** and
   **read** permissions to save new keys and retrieve existing
 
+#### Configuring Amazon KMS
+
+Amazon KMS can be used to encrypt and decrypt the passphrases that are used for
+encrypted RBD images. When a volume is created, a passphrase will be generated,
+which will be encrypted by the KMS and stored in the volumes metadata. Opon
+attaching the volume to a Pod, the worker node requests the KMS to decrypt the
+passphrase, after which it can be used to open the device with `cryptsetup` and
+provide access to it for the Pod.
+
+There are a few settings that need to be included in the [KMS configuration
+file](../examples/kms/vault/kms-config.yaml):
+
+1. `KMS_PROVIDER`: should be set to `aws-metadata`.
+1. `KMS_SECRET_NAME`: name of the Kubernetes Secret (in the Namespace where
+   Ceph-CSI is deployed) which contains the credentials for communicating with
+   AWS. This defaults to `ceph-csi-aws-credentials`.
+1. `AWS_REGION`: the region where the AWS KMS service is available.
+
+The [Secret with credentials](../examples/kms/vault/aws-credentials.yaml) for
+the AWS KMS is expected to contain:
+
+1. `AWS_ACCESS_KEY_ID`: ID of the key to use for encrypting/decrypting
+1. `AWS_SECRET_ACCESS_KEY`: secret for the key to use
+1. `AWS_SESSION_TOKEN`: *(optional)* session token, usually empty
+
+This Secret is expected to be created by the administrator who deployed
```

Comment:
can this be in any Namespace?

Reply:
No, it needs to be in the namesspace where Ceph-CSI is deployed.

Reply:
@nixpanic do we need to mention?

Reply:
No, I do not think so. If placing it in a different namespace was an option, *that* would need mentioning. Normally all objects for a deployment are located in the same (here ceph-csi) namespace.

Reply:
sure.